### PR TITLE
feat(ckb-indexer): support indexer rpc v0.4.1

### PR DESCRIPTION
**Pull request by PainterPuppets**
related task: https://github.com/nervosnetwork/lumos/issues/390

### feature:
- [x] support indexer rpc `get_cells`'s `with_data` options 
- [x] support indexer rpc `get_transaction`'s `group_by_transaction` options
- [x] support indexer rpc filter `script_len_range` options

### chore:
- [x] update `ckb-indexer` download script

### test case:
- [x] call `get_cells` with `with_data` option
- [x] call `get_transaction` with `group_by_transaction` option
- [x] call `get_cells` with `script_len_range` filter options
- [x] check `CellCollector` validate & convert scriptLenRange query

### CODE REVIEW

File: packages/ckb-indexer/tests/test_cases.js [hunk 1]
```diff
@@ -12,20 +12,22 @@ const lock = {
 const lockScriptWrapper = {
   script: {
     args: "0x7f2b32878434041acd17e396936de4f6ba377cce",
-    codeHash: "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
+    codeHash:
```
Comment (zhangyouxin) [inline packages/ckb-indexer/tests/test_cases.js]: Looks like that prettier is doing something unexpected

File: packages/ckb-indexer/src/collector.ts [hunk 2]
```diff
@@ -154,6 +158,10 @@ export class CKBCellCollector implements BaseCellCollector {
         ];
       }
     }
+
+    if (!this.queries.scriptLenRange && this.queries.type === "empty") {
```
Comment (zhangyouxin) [inline packages/ckb-indexer/src/collector.ts:162]: Not sure if this one is necessary, is there any error if we don't do this change?

**Comment by zhangyouxin** [comment]
Maybe we should add some test cases in `collector.e2e.test.ts` and `transaction_collector.e2e.test.ts` to test `collector.ts` and `transaction_collector.ts`.

File: packages/ckb-indexer/src/collector.ts (hunk 2 above)
Comment (homura) [reply packages/ckb-indexer/src/collector.ts:162]: > Not sure if this one is necessary, is there any error if we don't do this change?

this makes filter works in ckb-indexer

File: packages/ckb-indexer/tests/test_cases.js (hunk 1 above)
Comment (homura) [reply packages/ckb-indexer/tests/test_cases.js]: @PainterPuppets what has changed in this file? can you please ignore this file from .prettierrc?

**Comment by PainterPuppets** [comment]
> Maybe we should add some test cases in `collector.e2e.test.ts` and `transaction_collector.e2e.test.ts` to test `collector.ts` and `transaction_collector.ts`.

Actually `transaction_collector` and `collector` don't add `with_data` and `group_by_transactions` options, I'm unclear what aspect of the test should be added?

**Comment by zhangyouxin** [comment]
> Actually `transaction_collector` and `collector` don't add `with_data` and `group_by_transactions` options, I'm unclear what aspect of the test should be added?

OK I understand

**Review by homura — APPROVED** [review]

**Pull request merged**